### ISSUE TITLE
Multiple GeoIP databases?

### ISSUE THREAD

**Issue by alindt** [allinurl/goaccess#594]
It would be really helpful to allow multiple GeoIP databases to be specified (e.g. GeoLiteCity.dat _and_ GeoLiteCityv6.dat).

**Comment by solevi** [comment]
Hi,
This will be really helpful to have. I wonder how only two people requested that?

**Comment by stephenreay** [comment]
It seems this would also allow loading the GeoIP v4 and v6 databases concurrently? As I understand it @allinurl there's no way to do this currently?

**Comment by allinurl** [comment]
@stephenreay As far as I recall maxmind doesn't support this, but I'd need to take a look again and see if it's supported in the new lib.

**Comment by mdavids** [comment]
Any news on this?

**Comment by allinurl** [comment]
@mdavids I'm right i'm in the midst of replacing the on-disk storage, once I get that done (prob around March) which is long due, I'll be able to take a look at this and several other requests on my to-do list. Thanks for the heads up.

**Comment by ioannidesalex** [comment]
Any updates on this?

**Comment by allinurl** [comment]
Working on #117 as we speak. I'll bump this so it can get done after #117 is done. Thanks for the reminder.

**Comment by hardillb** [comment]
This appears to sort of work now

I've been able to add `--geoip-database` twice with both the GeoLite2-ASN and GeoLite2-City using goaccess v1.10.1 so I get both City level mapping and ASN data.

Only problem is it's not showing the table under the ASN chart or the world map to allow drilling down to specific country or region.

<img width="1610" height="1298" alt="Image" src="https://github.com/user-attachments/assets/5e418653-97aa-4c54-98a2-14ae4f36340e" />

and 

<img width="1636" height="796" alt="Image" src="https://github.com/user-attachments/assets/c58d7777-07a0-40e5-864a-e638b2f3be6a" />

Showing the following in the browser console

```
Uncaught TypeError: s.split is not a function
    hashCode https://www.hardill.me.uk/stats/:3
    renderRows https://www.hardill.me.uk/stats/:3
    renderDataRows https://www.hardill.me.uk/stats/:3
    renderFullTable https://www.hardill.me.uk/stats/:3
    renderTables https://www.hardill.me.uk/stats/:3
    initialize https://www.hardill.me.uk/stats/:3
    renderPanels https://www.hardill.me.uk/stats/:3
    initialize https://www.hardill.me.uk/stats/:3
    initializeWithoutWebSocket https://www.hardill.me.uk/stats/:3
    handleWebSocketSetup https://www.hardill.me.uk/stats/:3
    initialize https://www.hardill.me.uk/stats/:3
    onload https://www.hardill.me.uk/stats/:3
    EventHandlerNonNull* https://www.hardill.me.uk/stats/:3
    <anonymous> https://www.hardill.me.uk/stats/:3
```

Generated with

```
zcat /var/log/nginx/access.log.*.gz | /usr/local/bin/goaccess-1.10.1 /var/log/nginx/access.log /var/log/nginx/access.log.1 - -o /var/www/html/stats/index.html --log-format=VCOMBINED -j 2 -a --keep-last=14 --db-path=/root/stats --restore --persist --real-os --geoip-database=/usr/local/share/geoip/GeoLite2-ASN.mmdb --geoip-database=/usr/local/share/geoip/GeoLite2-City.mmdb
```

Happy to open another issue if you want and close this one as complete.

**Comment by allinurl** [comment]
@hardillb thanks for sharing that. Would you be able to attach the HTML file as well? Also, are you able to reproduce this with a smaller log or a subset of the original? If so, could you share that too? I could then try using that smaller subset to reproduce the issue on my end.

**Comment by hardillb** [comment]
OK, on a little bit more digging this appears to be a caused by an old AppPrefs Local storage object as I couldn't reproduce it with chrome compared to my usual firefox. Deleting the AppPrefs local storage in firefox appears to have fixed the problem.

I think you can close this whole issue out now

**Comment by allinurl** [comment]
@hardillb, thanks for reporting this! The issue should now be fixed. https://github.com/allinurl/goaccess/commit/b75a6d869f79914fc3b4ab009131108eef3eeea7

Feel free to build from [development](https://github.com/allinurl/goaccess?tab=readme-ov-file#build-from-github-development) or docker and let me know how it goes. 

It will be included in the next release.

**Comment by hardillb** [comment]
@allinurl I think I manually deleted the Local Storage object on all my machines, so not sure I can test this anymore, but thanks for sorting it.

**Comment by allinurl** [comment]
For the record, this is already supported via `--geoip-database`. Closing the issue for now, please reopen if needed.
